sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/noop/NoopSuite.scala

```diff
@@ -54,7 +56,7 @@ class NoopSuite extends SharedSQLContext {
           accum.add(1)
           x
         }
-        .write.format("noop").save()
+        .write.format("noop").mode("append").save()
```

Comment:
Can we change the default mode for v2 to `append` if no mode is set? That seems more reasonable than always requiring `mode("append")`.

Reply:
Personally I support it, but I'd like to do it in another PR, as we expect to have discussions about backward compatibility for changing the default mode.

Reply:
Okay, then I think this PR depends on the PR to make append the default mode for v2. Please get that in first.

Otherwise, this PR will make unnecessary changes to several places and we will have to roll those changes back to make append the default mode.

Reply:
what if we end up with not changing the default mode (e.g. the community doesn't agree with it)? do you think that's a blocker to forbid `ErrorIfExists` and `Ignore` mode for `TableProvider`?

Reply:
No, I don't think so. We would just have to make changes like this one. My motivation is to avoid making these changes because this touches several files just to set the mode to append.

Reply:
We can't change the default like this. It'd be a huge nightmare for users to upgrade, because it's a subtle semantic change. If we want to piss off all Spark users that write data pipelines, changing the default from one to another here is a great way to achieve that.

We can however introduce new APIs to make it more obvious what's happening.

Reply:
The proposal is only to change the default when using v2. If we don't change the default for v2, then the default is to always fail, which I think is worse.